src/rgw/rgw_tools.h

```diff
@@ -24,6 +24,8 @@ int rgw_init_ioctx(librados::Rados *rados, const rgw_pool& pool,
 
 int rgw_put_system_obj(RGWRados *rgwstore, const rgw_pool& pool, const string& oid, bufferlist& data, bool exclusive,
                        RGWObjVersionTracker *objv_tracker, real_time set_mtime, map<string, bufferlist> *pattrs = NULL);
+int rgw_put_system_obj(RGWRados *rgwstore, RGWSysObjectCtx& obj_ctx, const rgw_pool& pool, const string& oid, bufferlist& data, bool exclusive,
+                       RGWObjVersionTracker *objv_tracker, real_time set_mtime, map<string, bufferlist> *pattrs = NULL);
```

Comment:
since this overload isn't resolving the issue, can you please remove it? let's revisit RGWSysObjectCtx after more discussion